src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -628,6 +629,10 @@ def process_info_file(cp_elem, info_file):
 class ZincCompile(BaseZincCompile):
   """Compile Scala and Java code to classfiles using Zinc."""
 
+  deprecated_module(
```

Comment:
(...I'm actually not sure about this now, because this concrete class should not be extended by anyone? If you're not seeing it trigger on import of this file, then let's leave it I guess. Sorry!)

Reply:
Yea, it's still firing. Should remove or move it into the constructor as part of #8109.